### PR TITLE
[MERGE] website_blog, *: review blog layouts

### PR DESCRIPTION
**Pull request by kea14**
* website

Review the blog layouts to
- Use cards where necessary to match the forum / event redesign
- Use correct bootstrap / HTML

+ Some minor improvements

task-1948882

### CODE REVIEW

**Comment by kea14** [comment]
@qsm-odoo Before you take a look at the code, here is one question concerning the grid view (which works now but maybe not as expected). As I am now using bootstrap grid system and elements sometimes do not have the same height, this occurs : https://ibb.co/Xz7sKrW. Am I supposed to handle this with bootstrap (clearfix,... ?) or maybe with masonry or ... ? Thanks :smile:

**Comment by kea14** [comment]
@qsm-odoo I finally used Cards-columns directly from Bootstrap. You can check the code, everything seems good physically :smile:

**Comment by qsm-odoo** [comment]
@kea14 Is the order preserved with card-columns ?

**Comment by kea14** [comment]
@qsm-odoo Indeed... Order seems to be left to right : https://ibb.co/QMhvMy9 :disappointed:

**Comment by qsm-odoo** [comment]
As written in the doc ;) So it is not a viable solution. So I suppose there is only one solution: force the column to all have the same height (which should be the default when using the bootstrap grid system anyway). But I don't know, I have not investigated, do your best so that it looks good and the order is preserved :)

**Comment by kea14** [comment]
@qsm-odoo Final result for a blog post https://ibb.co/HqLnCzt Seems ok for you ? :)

**Comment by qsm-odoo** [comment]
Screenshot seems good :wink: Are you finished with the task or have you still something to improve ?

**Comment by kea14** [comment]
@qsm-odoo No, the task is done :)

File: addons/website_blog/views/website_blog_templates.xml [hunk 1]
```diff
@@ -406,146 +415,158 @@
...
 
 <!-- Option:Right Column: status -->
 <template id="opt_blog_rc_status" name="Status" priority="5"
         inherit_id="website_blog.index_right" active="True" customize_show="True">
     <xpath expr="//div[@id='main_column']" position="attributes">
-        <attribute name="class">col-md-8</attribute>
+        <attribute name="class">col</attribute>
     </xpath>
     <xpath expr="//div[@id='blog_right_column']" position="inside">
-        <section class="mt32 mb32" groups="website.group_website_designer">
```
Comment (qsm-odoo) [inline addons/website_blog/views/website_blog_templates.xml]: @kea14 I am working on this task. Comment about this (do not fix as I take care of this branch now): be careful, you removed a `groups` here, which means that all visitors would have had access to this panel :|

File: addons/website_blog/views/website_blog_templates.xml [hunk 2]
```diff
@@ -206,11 +211,15 @@
 <!-- Option: Blog Post List: grid layout -->
 <template id="opt_blog_post_grid_layout" name="Grid view"
         customize_show="True" active="False" inherit_id="website_blog.blog_post_short">
-    <xpath expr="//div[@id='main_column']" position="attributes">
-        <attribute name="class" separator=" " add="d-flex" />
+    <xpath expr="//div[@name='blog_post']" position="replace">
+        <div class="container">
```
Comment (qsm-odoo) [inline addons/website_blog/views/website_blog_templates.xml]: @kea14 A container cannot be inside of a col.

Bootstrap is: (element with padding or `container`) -> (content or (`row` -> `col`)) -> (content or (`row` -> `col`))  -> ...

File: addons/website_blog/views/website_blog_templates.xml [hunk 3]
```diff
@@ -380,20 +386,20 @@
...
-    <ul class="nav nav-pills ml16">
+    <ul class="nav nav-pills">
         <t t-if="tags">
             <div t-if='not hide_title' class='text-muted mt8'><i class='fa fa-tag' role="img" aria-label="Tag category" title="Tag category"/> <t t-esc='categ_title' /></div>
             <t t-foreach="tags" t-as="tag">
                 <t t-if="tag.post_ids">
-                    <li class="nav-item">
-                        <a t-attf-href="#{blog_url(tag=tags_list(active_tag_ids, tag.id))}" t-attf-class="nav-link#{tag.id in active_tag_ids and ' active' or ''}">
+                    <li class="nav-item mr-2">
+                        <a t-attf-href="#{blog_url(tag=tags_list(active_tag_ids, tag.id))}" t-attf-class="badge nav-link#{tag.id in active_tag_ids and ' active' or ''}">
```
Comment (qsm-odoo) [inline addons/website_blog/views/website_blog_templates.xml]: @kea14 General idea with bootstrap is composition: a component can contain others, but a component cannot be another. So a nav-link can contain a badge but not be a badge ;) If the result is good, this is just by pure luck ;) Then we switch to bootstrap 5... and the luck is gone... :)

File: addons/website_blog/views/website_blog_templates.xml [hunk 4]
```diff
@@ -406,146 +412,163 @@
...
-                </li>
-                <ul t-attf-class="blog_post_year collapse nav nav-pills flex-column #{not date and year_index == 0 and 'in'}" t-att-id="year" role="menu">
-                    <t t-foreach="nav_list[year]" t-as="months">
-                        <li t-if="months['date_begin'] == date" class="nav-item">
-                            <a role="menuitem" t-ignore="True" t-attf-href="#{blog_url(date_begin=False, date_end=False)}" class="nav-link active"><t t-esc="months['month']"/><span class="float-right badge badge-pill" t-esc="months['post_date_count']"/></a>
-                        </li>
-                        <li t-else="1" class="nav-item">
-                            <a role="menuitem" t-ignore="True" t-attf-href="#{blog_url(date_begin=months['date_begin'], date_end=months['date_end'])}" class="nav-link"><t t-esc="months['month']"/><span class="float-right badge badge-pill" t-esc="months['post_date_count']"/></a>
+        <div class="card mb-3 bg-secondary">
+            <h6 class="card-header"><b>Archives</b></h6>
```
Comment (qsm-odoo) [inline addons/website_blog/views/website_blog_templates.xml]: @kea14 You cannot use `<b>` to make titles bold, this the responsibility of the theme to say if titles are bold, so you should not force them with HTML elements.

File: addons/website_blog/views/website_blog_templates.xml [hunk 5]
```diff
@@ -406,146 +412,163 @@
...
     </xpath>
     <xpath expr="//div[@id='blog_right_column']" position="inside">
-        <section class="mt32">
-        <h4>About us</h4>
-        <p>
-            Write a small text here for when <b>new visitors</b> find your website
-            through your <b>blog entries</b>, referenced in Google.
-        </p>
-        </section>
+        <div class="card mb-3 bg-secondary" id="about_forum">
```
Comment (qsm-odoo) [inline addons/website_blog/views/website_blog_templates.xml]: @kea14 about_forum ?

**Comment by qsm-odoo** [comment]
@robodoo r+

**Comment by qsm-odoo** [comment]
@robodoo rebase-merge

**Pull request closed without merging**